pkg/kubelet/kubelet.go

```diff
@@ -2202,19 +2208,23 @@ func (kl *Kubelet) updateRuntimeUp() {
 
 	s, err := kl.containerRuntime.Status()
 	if err != nil {
-		klog.Errorf("Container runtime sanity check failed: %v", err)
+		klog.ErrorS(err, "Container runtime sanity check failed")
 		return
 	}
 	if s == nil {
-		klog.Errorf("Container runtime status is nil")
+		klog.ErrorS(nil, "Container runtime status is nil")
 		return
 	}
 	// Periodically log the whole runtime status for debugging.
-	klog.V(4).Infof("Container runtime status: %v", s)
+	klog.V(4).InfoS("Container runtime status", "status", s)
 	networkReady := s.GetRuntimeCondition(kubecontainer.NetworkReady)
 	if networkReady == nil || !networkReady.Status {
-		klog.Errorf("Container runtime network not ready: %v", networkReady)
-		kl.runtimeState.setNetworkState(fmt.Errorf("runtime network not ready: %v", networkReady))
+		if networkReady != nil {
+			klog.ErrorS(nil, "Container runtime network not ready", "networkReady", networkReady.String())
+		} else {
+			klog.ErrorS(nil, "Container runtime NetworkReady condition not set")
+		}
```

Comment:
`.String()` requires additional `nil` check for the condition, should we have a helper in `klog` (like `KRef`) to take care of this and avoid such nested ifelse ? cc @serathius @ehashman

Reply:
I think this is fine since we're logging a different error message depending on the case

Reply:
Does klog not handle the case where `networkReady=nil`? I thought it does. We shouldn't have to call .String() here at all

Reply:
yes, removed the `.String()` and kept the logging for difference case